src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/ExecuteTaskReconfigurationServiceTest.java

```diff
@@ -0,0 +1,78 @@
+package uk.gov.hmcts.reform.wataskmanagementapi.services;
+
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.jdbc.Sql;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import uk.gov.hmcts.reform.wataskmanagementapi.cft.repository.TaskResourceRepository;
+import uk.gov.hmcts.reform.wataskmanagementapi.controllers.request.TaskOperationRequest;
+import uk.gov.hmcts.reform.wataskmanagementapi.controllers.request.entities.ExecuteReconfigureTaskFilter;
+import uk.gov.hmcts.reform.wataskmanagementapi.controllers.request.entities.TaskFilter;
+import uk.gov.hmcts.reform.wataskmanagementapi.controllers.request.entities.TaskOperation;
+import uk.gov.hmcts.reform.wataskmanagementapi.controllers.request.enums.TaskFilterOperator;
+import uk.gov.hmcts.reform.wataskmanagementapi.controllers.request.enums.TaskOperationName;
+import uk.gov.hmcts.reform.wataskmanagementapi.taskconfiguration.services.ConfigureTaskService;
+import uk.gov.hmcts.reform.wataskmanagementapi.taskconfiguration.services.TaskAutoAssignmentService;
+
+import java.time.Duration;
+import java.time.OffsetDateTime;
+import java.util.List;
+
+@ActiveProfiles("integration")
+@DataJpaTest
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
+@Testcontainers
+@Sql("/scripts/wa/reconfigure_task_data.sql")
+public class ExecuteTaskReconfigurationServiceTest {
+    private CFTTaskDatabaseService cftTaskDatabaseService;
+    @MockBean
+    private ConfigureTaskService configureTaskService;
+    @MockBean
+    private TaskAutoAssignmentService taskAutoAssignmentService;
+    @Autowired
+    TaskResourceRepository taskResourceRepository;
+    private ExecuteTaskReconfigurationService executeTaskReconfigurationService;
+
+    @BeforeEach
+    void setUp() {
+        cftTaskDatabaseService = new CFTTaskDatabaseService(taskResourceRepository);
+        executeTaskReconfigurationService = new ExecuteTaskReconfigurationService(
+            cftTaskDatabaseService,
+            configureTaskService,
+            taskAutoAssignmentService);
+    }
+
+    @Test
+    void should_get_reconfiguration_fail_log() {
+        List<TaskFilter<?>> taskFilters = createReconfigureTaskFilters();
+
+        TaskOperationRequest taskOperationRequest = new TaskOperationRequest(
+            TaskOperation.builder()
+                .name(TaskOperationName.EXECUTE_RECONFIGURE)
```

Comment:
could you please another tests for MARK_TO_RECONFIGURE and invalid task operation type test?